src/libraries/System.Private.CoreLib/src/System/Double.cs

```diff
@@ -440,6 +440,163 @@ public static bool TryParse(ReadOnlySpan<char> s, NumberStyles style, IFormatPro
             return Number.TryParseFloat(s, style, NumberFormatInfo.GetInstance(provider), out result);
         }
 
+        //
+        // Helpers, those methods are referenced from the JIT
+        //
+
+        private const double IntMaxValueOffset = 2147483648.0; // 2^31, int.MaxValue + 1
+        private const double UIntMaxValueOffset = 4294967296.0; // 2^32, uint.MaxValue + 1
+
+        // TODO: make the helpers saturating when implementing https://github.com/dotnet/runtime/issues/61885
+        private static int ConvertToInt32(double value) => (int)(long)value;
+        private static uint ConvertToUInt32(double value) => (uint)(long)value;
```

Comment:
What is the performance of this implementation compared to RhpDbl2Int/RhpDbl2UInt that was used before this change?

Reply:
`RhpDbl2Int` is unused today AFAIK and `RhpDbl2UInt` will only be used by X86 which @filipnavara is currently exposing so there'd be no observable regression.
As such, I'd rather keep things simple here and make it slightly slower for NAOT X86 due to the additional call and let it get fixed up with either helper inlining or the Intel saturating changes. Speaking of those, changing it to a managed implementation in NAOT makes it so that they don't have to test changes to a C++ helper that's only used on a non fully supported platform.

Reply:
I would like to see the impact of these changes to be measured, or the changes to be reverted.

Reply:
Baseline: 573
PR: 630

for this code:
```csharp
using System;
using System.Diagnostics;
using System.Runtime.CompilerServices;

for (;;)
{
    Stopwatch sw = new();
    sw.Start();
    Work(10_000_000_000);
    Console.WriteLine(sw.ElapsedMilliseconds.ToString());
}

[MethodImpl(MethodImplOptions.NoInlining)]
static uint Work(double iters)
{
    uint sum = 0;
    for (double i = 0; i < iters; i += 42)
    {
        sum += (uint)i;
    }
    return sum;
}
```

Reply:
I do not see a good reason for taking this regression.